alexandria/core/views.py

```diff
@@ -114,6 +116,35 @@ def update(self, request, *args, **kwargs):
 
         return response
 
+    @action(
+        methods=["post"],
+        detail=True,
+        url_path="copy",
+    )
+    def copy(self, request, pk=None):
+        document = self.get_object()
+        user, group = import_string(settings.ALEXANDRIA_GET_USER_AND_GROUP_FUNCTION)(
+            request
+        )
+
+        json_body = json.loads(request.body.decode("utf-8"))
+        category_id = json_body.get("data", {}).get("category")
+        if category_id:
+            category = get_object_or_404(models.Category, pk=category_id)
+        else:
+            category = document.category
+
+        copied_document, __ = copy_document(
+            document=document,
+            category=category,
+            user=user,
+            group=group,
+        )
```

Comment:
Could we maybe move this into a specialized serializer instead? This seems like a lot of manual work dealing with the request here. Or, at least, there should be `request.data` I think, which would help avoid dealing with JSON

Reply:
I created a new specialized serializer to handle the request input, and convert the category parameter into a category object when provided. It now also uses `request.data` to not deal with the JSON anymore.